drivers/can/can_sam.c

```diff
@@ -101,7 +102,13 @@ static int can_sam_init(const struct device *dev)
 		return ret;
 	}
 
-	ret = can_mcan_configure_mram(dev, 0U, sam_cfg->mram);
+	/* get actual message ram base address */
+	uint32_t mrba = sam_cfg->mram & 0xFFFF0000;
+
+	/* keep lower 16bit; update DMA Base Address */
+	*sam_cfg->dma_base_addr = (*sam_cfg->dma_base_addr & 0x0000FFFF) | mrba;
```

Comment:
Please change this to use `sys_write32()` for writing the register (similar to all other register writes done in the M_CAN backend driver.